feat(#2594): document link component

bdfranck opened the pull request:
This PR adds documentation for the Link component which now works with our `alpha` Angular and React components. (Implemented in https://github.com/GovAlta/ui-components/issues/2522)

twjeffery commented on src/routes/components/Link.tsx:
```diff
@@ -0,0 +1,126 @@
+        name="Link"
+        category={Category.UTILITIES}
+        description="Wraps an anchor element to add icons or margins."
+      />
```
Also add related components:  
```
relatedComponents={[
          { link: "/components/button", name: "Button" },
        ]}
```

twjeffery commented on src/routes/components/Link.tsx:
```diff
@@ -0,0 +1,126 @@
+          heading={
+            <>
+              Design guidelines
+              <GoabBadge type="information" content="In progress" />
```
remove "in progress" badge

twjeffery commented on src/routes/components/Link.tsx:
```diff
@@ -0,0 +1,126 @@
+        <GoabTab
+          heading={
+            <>
+              Design guidelines
```
rename to "Design"

bdfranck commented:
@twjeffery Thanks for the review! I've amended my commit with the following changes:

- Added the Figma link and GitHub link to the appropriate locations
- Added the related component section
- Changed the label of the design tab to "Design"
- Removed the badge from the design tab label
- Added the accessibility tab
- Added the new empty state for design and accessibility

twjeffery reviewed (approved):
Changes look great, approved by me
